src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php

```diff
@@ -55,9 +55,9 @@ private function locateFilesInFilesInFilesDefinitions(array $locations) : Genera
         foreach ($locations as $location) {
             if (is_file($location)) {
                 yield $location;
+            } elseif (is_dir($location)) {
+                yield from $this->extractFilesFromDirectory($location);
             }
-
-            yield from $this->extractFilesFromDirectory($location);
```

Comment:
Nope: Testing ```is_file``` and ```is_dir``` check whether the given items are a regular file or a directory. 

```
$ touch foo
$ mkdir bar
$ ln -s /dev/urandom foobar
$ ls -al
total 8
drwxr-xr-x   5 heiglandreas  wheel  170 23 Jan 16:56 .
drwxrwxrwt  16 root          wheel  544 23 Jan 16:55 ..
drwxr-xr-x   2 heiglandreas  wheel   68 23 Jan 16:55 bar
-rw-r--r--   1 heiglandreas  wheel    0 23 Jan 16:55 foo
lrwxr-xr-x   1 heiglandreas  wheel   12 23 Jan 16:56 foobar -> /dev/urandom
$ pwd
/tmp/linktest
$ php -a
Interactive shell

php > var_Dump(is_dir('/tmp/linktest/foobar'));
bool(false)
php > var_Dump(is_file('/tmp/linktest/foobar'));
bool(false)
php > var_Dump(is_link('/tmp/linktest/foobar'));
bool(true)
```